pkg/analysis/passes/backenddebug/backenddebug.go

```diff
@@ -0,0 +1,56 @@
+package backenddebug
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/grafana/plugin-validator/pkg/analysis"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/archive"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/metadata"
+	"os"
+	"path/filepath"
+)
+
+var (
+	backendDebugFilePresent = &analysis.Rule{Name: "backend-debug-file-present", Severity: analysis.Error}
+)
+
+// Analyzer is an analyzer that checks if backend standalone debug files are included in the executable.
+// If so, it reports an error, as the plugin can't be used properly in non-debug mode.
+var Analyzer = &analysis.Analyzer{
+	Name:     "backenddebug",
+	Requires: []*analysis.Analyzer{archive.Analyzer, metadata.Analyzer},
+	Run:      run,
+	Rules:    []*analysis.Rule{backendDebugFilePresent},
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	archiveDir := pass.ResultOf[archive.Analyzer].(string)
```

Comment:
`archiveDir, ok` := pass.Res and check for `ok` or `return nil,nil`. Same with the line below